### PR TITLE
Fix rust-bio-tools BLAS linking

### PR DESCRIPTION
**Pull request by mbargull**
When we link to BLAS libraries, we should use the generic `libblas`/`libcblas`/etc. dependencies which during build default to their "netlib" variants. This way, our binaries are linked to libraries with a common ABI such that downstream any compatible BLAS implementation can be used (e.g., the user can choose to install `mkl`/`blis`/`openblas` on their own by installing `libcblas * *mkl` or similar).
For more details, see https://conda-forge.org/docs/maintainer/knowledge_base.html#blas

The recipe in its old form used `libopenblas` (and `blis`??) which causes linking to specific BLAS implementations. Furthermore, it does not guarantee that, e.g., `libopenblas` is linked to since another BLAS implementation, e.g., `mkl`, could be pulled in via indirect dependencies (likely via `gsl` in this case). This can happen since `libopenblas` and `mkl` can be installed side by side. To avoid this, the `libcblas` (& co.) packages act as "mutex" packages, i.e., only one `libcblas` package may be installed at a time, e.g., adding a dependency on `libcblas * *openblas` will prevent `libcblas * *mkl` to be installed. Plus, as mentioned before, `libcblas * *netlib` will be chosen by default during our builds such that we avoid linking to specific implementations.

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/rust-bio-tools/build.sh [hunk 1]
```diff
@@ -1,18 +1,14 @@
...
-    mkdir -p $HOME/.nim
-
-    # according to https://github.com/rust-lang/cargo/issues/2422#issuecomment-198458960 removing circle ci default configuration solves cargo trouble downloading crates
-    #git config --global --unset url.ssh://git@github.com.insteadOf
-
-fi
-
-# build statically linked binary with Rust
-C_INCLUDE_PATH=$PREFIX/include LIBRARY_PATH=$PREFIX/lib cargo install --path . --root $PREFIX
+# Dependency crate librocksdb-sys uses SSE4.1/4.2 and PCLMUL unconditionally.
```
Comment (mbargull) [inline recipes/rust-bio-tools/build.sh:3]: ref: https://github.com/rust-rocksdb/rust-rocksdb/pull/295

**Comment by mbargull** [comment]
@BiocondaBot, please fetch artifacts

**Comment by mbargull** [comment]
@BiocondaBot, please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**